sqlglot/dialects/duckdb.py

```diff
@@ -548,3 +568,18 @@ def join_sql(self, expression: exp.Join) -> str:
                 return super().join_sql(expression.on(exp.true()))
 
             return super().join_sql(expression)
+
+        def generateseries_sql(self, expression: exp.GenerateSeries) -> str:
+            is_end_exclusive = expression.args.get("is_end_exclusive")
+
+            # DuckDB has the following functions:
+            # - Generate_Series(a, b) -> generates [a, b]
+            # - Range(a, b) -> generates [a, b)
+            if is_end_exclusive:
+                start = expression.args.get("start")
+                end = expression.args.get("end")
+                step = expression.args.get("step")
+
+                return self.func("RANGE", start, end, step)
```

Comment:
```suggestion
            # GENERATE_SERIES(a, b) -> [a, b], RANGE(a, b) -> [a, b)
            if expression.args.get("is_end_exclusive"):
                expression.set("is_end_exclusive", None)
                return rename_func("RANGE")(self, expression)
```

Reply:
You're making a very valid point, but the `Generator` [copies ASTs by default](https://github.com/tobymao/sqlglot/blob/2a88e40da89fa083bbd8fd0174082fa8e677780a/sqlglot/generator.py#L542), so mutations aren't a concern.